### ISSUE TITLE
GP's desk placement different than original gameplay.

### ISSUE THREAD

**Issue by DoctorToboggan** [CorsixTH/CorsixTH#602]
What steps will reproduce the problem?
1.  Running the original and Corsix back to back for comparison.
2.  On any level, build a 4x4 GP's office, place the door on the rightmost tile of the bottom right side, with a window next to it.
3.  Attempt to place the desk in the bottom left corner.

What is the expected output? What do you see instead?
The desk should highlight and be able to be put there.  It stays greyed out and cannot be placed instead.

What version of CorsixTH are you using (e.g. "0.30", "4c66e39985")?
  Version 0.40, release date 22th December 2014

What operating system / compile settings are you using?
I'm using Windows Ultimate SP1 64-bit, running CorsixTH from the installer located here: https://github.com/CorsixTH/CorsixTH/releases  My Theme Hospital files are from a DOSBox 0.74 install of the game using the CD-ROM's dos setup, I added the intro file manually.  I don't know if the game was patched to 1.1, but Corsix never complained about the level/data files other than the intro.

What level was this on (e.g. "Demo level", "Full game level 12")?
Level 1

Corsix: ![th-corsixth](https://cloud.githubusercontent.com/assets/10442081/5655328/57d1f40a-9692-11e4-8f52-03a6c9ca17a6.jpg)

Original/DOSBox: ![th-dosbox](https://cloud.githubusercontent.com/assets/10442081/5655336/768b38ca-9692-11e4-96ad-33848fb13d17.jpg)

**Comment by MarkL1961** [comment]
I have seen this as well.  
The x-ray room can at times be impossible to place both objects at some sizes, depending on where the door is.  I can think of one layout where it won't allow you to place the shield against the wall, but will let you one tile away from the wall, which I think is odd.

**Comment by TheCycoONE** [comment]
Maybe the original allowed inaccessible tiles in a room as long as there was no need to enter there, and CorsixTH doesn't.  What happens if there is a plant in that back corner?

**Comment by DoctorToboggan** [comment]
I'll do some more testing and see what happens.  I usually shove the fire extinguisher back there, cause safety and all that.  I know off my head that using that 4x4 with the door where it is, that is the only place the desk can go, but isn't the only 4x4 that works.

**Comment by mugmuggy** [comment]
Can replicate the issue. As @DoctorToboggan indicates, placing a plant back there it dies off in the original. Putting a footprint tile of {-2, 0, invisible = true} in north and {0, -2, invisible = true} in west orientations for the desk seems to do the trick. The other orientations already cover 6 tiles, but have 2 as passable each and from a quick look align with TH.

The issues stems from World:wouldNonSideObjectBreakPathfindingIfSpawnedAt failing on the extra tile as its not part of the footprint therefore its adjacent and would need to be connected to the rest of the hospital.

**Comment by MarkL1961** [comment]
There is or was a rule to prevent blocked off areas in TH. (discussed in another issue I think)

**Comment by TheCycoONE** [comment]
https://github.com/CorsixTH/CorsixTH/issues/826 - Did rooms normally allow blocking off equipment? I feel like this was an area where the path finding was actually used, but I haven't tested yet.

**Comment by MarkL1961** [comment]
Yes you could stick things in a blocked corner as stated above.  Plants though would die because the handyman could only get near a plant if the room was not used, if at all.  This though could have been a bug in TH.  I never used to place many plants in rooms anyway, but I can remember putting the fire extinguisher in a corner like that because once placed there was no interaction.
